akka-cluster/src/main/scala/akka/cluster/ClusterDaemon.scala

```diff
@@ -1120,6 +1127,11 @@ private[cluster] class ClusterCoreDaemon(publisher: ActorRef, joinConfigCompatCh
         coordShutdown.run(CoordinatedShutdown.ClusterLeavingReason)
       }
 
+      if (selfStatus == Down && localGossip.member(selfUniqueAddress).status != Down) {
```

Comment:
Note that MemberDowned event will be published immediately when receiving self as Down. ClusterSingleton and ClusterSharding subscribe to this event and will immediately stop themselves even though the ActorSystem is running some longer.